### ISSUE TITLE
Suggested replacement for addIgnoreFile doesn't work always

### ISSUE THREAD

**Issue by dhilgarth** [kaelzhang/node-ignore#108]
With the attached .gitignore, the following code will return `false`

```
const ig = ignore.default();
ig.add(fs.readFileSync('.gitignore').toString());
console.log(ig.ignores('node_modules'));
```

I believe it's because of the BOM at the beginning of the file.

[gitignore.zip](https://github.com/kaelzhang/node-ignore/files/14065065/gitignore.zip)

**Comment by kaelzhang** [comment]
confirmed

**Comment by kaelzhang** [comment]
I will do a complete tests for all zero-with chars

**Comment by kaelzhang** [comment]
Landed in `5.3.1`

**Comment by dhilgarth** [comment]
Awesome, thanks!
